keel-core/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactVersionLinks.kt

```diff
@@ -51,22 +52,40 @@ class ArtifactVersionLinks(private val scmInfo: ScmInfo, private val cacheFactor
     } else null
   }
 
+  fun getScmType(commitLink: String): String? {
+    val commitURL = URL(commitLink)
+
+    val scmBaseURLs = runBlocking {
+      val cachedValue = cache[cacheName].get()
+      cachedValue ?: scmInfo.getScmInfo()
+    }
+
+    val base = scmBaseURLs.filter { (_,baseUrl) ->
+      commitURL.host == URL(baseUrl).host
+    }
+
+    return base.keys.toList()[0]
+  }
+
   //Calling igor to fetch all base urls by SCM type, and returning the right one based on current commit link
   fun getScmBaseLink(commitLink: String): String? {
-
+    val scmType = getScmType(commitLink)
     val scmBaseURLs = runBlocking {
       val cachedValue = cache[cacheName].get()
       cachedValue ?: scmInfo.getScmInfo()
     }
-    return when {
-      "stash" in commitLink ->
-        scmBaseURLs["stash"]
-      "github" in commitLink -> {
-        val url = URL(scmBaseURLs["gitHub"])
-        "${url.protocol}://${url.host}"
+
+    return if (scmType != null) {
+      when {
+        "stash" in scmType ->
+          scmBaseURLs["stash"]
+        "gitHub" in scmType -> {
```

Comment:
same here. problably better if `getScmType` converts the type to lower case to ensure consistency.

Reply:
I would rather perform the normalization of the string closer to the site of the check to avoid ambiguity. Check my latest commits.